### PR TITLE
Feature: Cobra / Viper config

### PR DESCRIPTION
**Pull request by scruplelesswizard**
Use Cobra / Viper for configuration, allowing for use of config files and/or command line flags.

Important for kubermatic/config#78

### CODE REVIEW

File: cmd/kubermatic-api/cmd/root.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+	"github.com/kubermatic/api/handler"
+	"github.com/kubermatic/api/provider"
+	"github.com/kubermatic/api/provider/cloud"
+	"github.com/kubermatic/api/provider/kubernetes"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+)
+
+var cfgFile, kubeConfig, dcFile, secretsFile, jwtKey, address string
```
Comment (sttts) [inline cmd/kubermatic-api/cmd/root.go:36]: Am not a big fan of global variables. At the very least I would put them into a Config struct.

File: cmd/kubermatic-api/cmd/root.go [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+// Execute adds all child commands to the root command sets flags appropriately.
+// This is called by main.main(). It only needs to happen once to the rootCmd.
+func Execute() {
+	if err := RootCmd.Execute(); err != nil {
+		fmt.Println(err)
+		os.Exit(-1)
+	}
+}
+
+func init() {
```
Comment (sttts) [inline cmd/kubermatic-api/cmd/root.go:92]: Is this the normal pattern to use init funcs to setup the cobra command?

Comment (scruplelesswizard) [reply cmd/kubermatic-api/cmd/root.go:92]: From the cobra docs, and from minikube I would say yes

File: cmd/kubermatic-api/cmd/root.go (hunk 1 above)
Comment (scruplelesswizard) [reply cmd/kubermatic-api/cmd/root.go:36]: Lets open an issue to move them out after we get multi-seed running

File: cmd/kubermatic-api/cmd/root.go [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+	// If a config file is found, read it in.
+	if err := viper.ReadInConfig(); err == nil {
+		fmt.Println("Using config file:", viper.ConfigFileUsed())
+	}
+}
+
+func setFlagsUsingViper() {
+	for _, config := range viperWhiteList {
+		var a = pflag.Lookup(config)
```
Comment (scheeles) [inline cmd/kubermatic-api/cmd/root.go]: Can we use a clear name for `a`

File: cmd/kubermatic-api/cmd/root.go [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+	Use:   "api",
+	Short: "Kubermatic API Server",
+	Long:  `API server for Kubermatic, providing access to seed and client resources within the clusters`,
+	// Uncomment the following line if your bare application
+	// has an action associated with it:
+	Run: func(cmd *cobra.Command, args []string) {
+
+		// load list of datacenters
+
+		dcs := map[string]provider.DatacenterMeta{}
```
Comment (scheeles) [inline cmd/kubermatic-api/cmd/root.go:54]: Change variables names, Luk start with it already see https://github.com/kubermatic/api/pull/97/files

Comment (scruplelesswizard) [reply cmd/kubermatic-api/cmd/root.go:54]: This should be done on the documentation branch I think...

**Review by scheeles — APPROVED** [review]
LGTM

**Pull request merged**